deckhouse-controller/pkg/controller/module-controllers/utils/utils.go

```diff
@@ -127,3 +133,60 @@ func ParseDeckhouseRegistrySecret(data map[string][]byte) (*DeckhouseRegistrySec
 		CA:                    string(ca),
 	}, err
 }
+
+func Update[Object client.Object](ctx context.Context, cli client.Client, object Object, status bool, updater func(obj Object) bool) error {
+	return retry.OnError(retry.DefaultRetry, apierrors.IsServiceUnavailable, func() error {
+		return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			if err := cli.Get(ctx, client.ObjectKey{Name: object.GetName()}, object); err != nil {
+				return err
+			}
```

Comment:
This function allows us to avoid conflict cases. We need to get a 'fresh' object before updating it. Yes, if updater returns false, getting is useless :(, but I have no idea how to redesign it, and due to the cache, this operation is not expensive.

Reply:
but then, why don't we get an object every time before updating in the current implementations?

Reply:
In the current implementation, we were not afraid of conflict. Bu solution is just a precaution, if you think it's unnecessary, I can get rid of it.

Reply:
I understand and backup your incentive, I'm merely trying to remember if it's safe to do whatever you want with an object or not. Give me please some time to figure it out, I'll get back to you later.

Reply:
It seems you're right, even when using a cache, we still have to have an up-to-date object if we want to update it successfully. Though,  if we have to update frequently and multiple fields, maybe we could give `patch` a try as it cares about `resourceVersion` and etc.